presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -405,7 +403,7 @@ public PlanOptimizers(
                 new CheckSubqueryNodesAreRewritten(),
                 new StatsRecordingPlanOptimizer(
                         optimizerStats,
-                        new PredicatePushDown(metadata, typeAnalyzer, false)),
+                        new StatsRecordingPlanOptimizer(optimizerStats, new PredicatePushDown(metadata, typeAnalyzer, false, false))),
```

Comment:
Since we are using a new instance of `StatsRecordingPlanOptimizer` for the 3 places where `PredicatePushDown` rule is applied, it looks like this will reset the stats each time for this rule (Check `OptimizerStatsRecorder#register`). Is that ok ? I think we would want to know the combined time taken of `PredicatePushDown` rule in all the invocations or record them all separately.

Reply:
I will report to same `optimizerStats` every time, so it accumulates stats for `PredicatePushDown`